src/librbd/io/ImageRequest.h

```diff
@@ -308,6 +311,31 @@ class ImageWriteSameRequest : public AbstractImageWriteRequest<ImageCtxT> {
   int m_op_flags;
 };
 
+template <typename ImageCtxT = ImageCtx>
+class ImageCmpextRequest : public ImageRequest<ImageCtxT> {
+public:
+  using typename ImageRequest<ImageCtxT>::Extents;
+
+  ImageCmpextRequest(ImageCtxT &image_ctx, AioCompletion *aio_comp,
+                   Extents &&image_extents, bufferlist &&cmp_bl);
+
+protected:
+  void send_request() override;
+  void send_image_cache_request() override {
+    //TODO
+    return;
```

Comment:
Nit: nothing is using the image cache right now, so I'd prefer to just "assert(false)" here to make future debugging easier.